src/main/java/org/folio/marc4ld/service/ld2marc/processing/combine/impl/DataFieldCombinerFactoryImpl.java

```diff
@@ -1,23 +1,36 @@
 package org.folio.marc4ld.service.ld2marc.processing.combine.impl;
 
+import static org.folio.ld.dictionary.ResourceTypeDictionary.DISSERTATION;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.INSTANCE;
+
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
 import org.folio.marc4ld.service.ld2marc.processing.combine.DataFieldCombiner;
 import org.folio.marc4ld.service.ld2marc.processing.combine.DataFieldCombinerFactory;
 import org.marc4j.marc.DataField;
-import org.springframework.stereotype.Component;
 
-@Component
 public class DataFieldCombinerFactoryImpl implements DataFieldCombinerFactory {
 
+  private final Map<Predicate<DataFieldCombiner.Context>, DataFieldCombiner> combinerMap;
+
+  public DataFieldCombinerFactoryImpl() {
+    combinerMap = Map.of(
+      c -> c.marcTag().equals("245") && c.resourceTypes().contains(INSTANCE), new TitleCombiner(),
+      c -> c.marcTag().equals("010") && c.resourceTypes().contains(INSTANCE), new LccnCombiner(),
+      c -> c.marcTag().equals("502") && c.resourceTypes().contains(DISSERTATION), new DissertionCombiner()
```

Comment:
Minor.
For me it's a separate class, like:
```
DissertionCombinerFactory
LccnCombinerFactory
```
with interface
`CombinerFactory implements Predicate<DataFieldCombiner.Context>, Supplier<DataFieldCombiner>` {
    
String getTag() // unnecessary but for optimisation to create Map<String, CombinerFactory>
}

Reply:
Thanks @aleksei-pronichev. I merged your commit to this brach.